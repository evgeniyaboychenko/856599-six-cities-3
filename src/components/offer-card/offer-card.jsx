import React from 'react';
import PropTypes from 'prop-types';


const getPercent = (rating) => {
  return rating * 100 / 5;
};

const getPremium = (isPremium) => {
  if (isPremium) {
    return (
      <div className="place-card__mark">
        <span>Premium</span>
      </div>);
  }
  return ``;
};

const OfferCard = (props) => {
  const {offerCard, onHeaderCardClick, onCardMouseover, cardType} = props;
  const {id, photos, name, rating, price, type, isPremium} = offerCard;
  return (
    <article className={cardType + `__place-card place-card`} key = {id} onMouseOver = {(evt) => {
      evt.preventDefault();
      onCardMouseover(id);
    }
    }>
      {getPremium(isPremium)}
      <div className={cardType + `__image-wrapper place-card__image-wrapper`}>
        <a href="#">
          <img className="place-card__image" src = {`img/` + photos[0]} width="260" height="200" alt="Place image"/>
        </a>
      </div>
      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>
          <button className="place-card__bookmark-button place-card__bookmark-button--active button" type="button">
            <svg className="place-card__bookmark-icon" width="18" height="19">
              <use href="#icon-bookmark"/>
            </svg>
            <span className="visually-hidden">In bookmarks</span>
          </button>
        </div>
        <div className="place-card__rating rating">
          <div className="place-card__stars rating__stars">
            <span style={{width: getPercent(rating)}}></span>
            <span className="visually-hidden">Rating</span>
          </div>
        </div>
        <h2 className="place-card__name">
          <a href="#" onClick = {(evt) => {
            evt.preventDefault();
            onHeaderCardClick(id);
          }
          }>
            {name}
          </a>
        </h2>
        <p className="place-card__type">{type}</p>
      </div>
    </article>);
};

OfferCard.propTypes = {
  offerCard: PropTypes.shape({
    id: PropTypes.string.isRequired,
    photos: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    name: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    price: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    isPremium: PropTypes.bool.isRequired
  }),
  onHeaderCardClick: PropTypes.func.isRequired,
  onCardMouseover: PropTypes.func.isRequired,
  cardType: PropTypes.string.isRequired,
};

export default OfferCard;
