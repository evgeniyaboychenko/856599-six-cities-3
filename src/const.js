const CardType = {
  NEAR: `near-places`,
  CITY: `cities`,
  FAVORITES: `favorites`
};

const SortType = {
  DEFAULT: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`
};

const AppRoute = {
  SIGN_IN: `/login`,
  FAVORITES: `/favorites`,
  ROOM: `/offer/:id`,
  MAIN: `/`,
};

const ResponseErrorCode = {
  UNAUTHORIZED: 401,
  INVALID_REQUEST: 400,
  NOT_FOUND: 404,
  SERVER_IS_NOT_AVAILABLE: 500
};

export {CardType, SortType, AppRoute, ResponseErrorCode};
