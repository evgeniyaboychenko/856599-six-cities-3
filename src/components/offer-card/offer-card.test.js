import React from 'react';
import renderer from 'react-test-renderer';
import {OfferCard} from './offer-card.jsx';
import {CardType} from '../../const.js';
import {MemoryRouter} from "react-router-dom";

const OFFER_IMAGES = [`room.jpg`, `apartment-01.jpg`];
const OFFER_NAMES = [`Canal View Prinsengracht`, `Nice, cozy, warm big bed apartment`];
const OFFER_TYPES = [`apartment`, `privet room`];

const offerCard =
{
  id: `1`,
  photos: OFFER_IMAGES,
  name: OFFER_NAMES[0],
  rating: 2,
  price: 100,
  type: OFFER_TYPES[0],
  isPremium: true,
};

describe(`OfferCard component render correctly`, () => {
  it(`should OfferCard render correctly for offer cities`, () => {
    const tree = renderer.create(
        <MemoryRouter>
          <OfferCard
            offerCard = {offerCard}
            // onHeaderCardClick = {() => {}}
            onCardMouseover = {() => {}}
            cardType = {CardType.CITY}
          />
        </MemoryRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should OfferCard render correctly for offer near`, () => {
    const tree = renderer.create(
        <MemoryRouter>
          <OfferCard
            offerCard = {offerCard}
            // onHeaderCardClick = {() => {}}
            onCardMouseover = {() => {}}
            cardType = {CardType.NEAR}
          />
        </MemoryRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
