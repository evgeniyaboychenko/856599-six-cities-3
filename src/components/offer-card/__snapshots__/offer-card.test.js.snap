// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard component render correctly should OfferCard render correctly for offer cities 1`] = `
<article
  className="cities__place-card place-card"
  onMouseOver={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="/img/room.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          100
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            href="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": 40,
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        Canal View Prinsengracht
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;

exports[`OfferCard component render correctly should OfferCard render correctly for offer near 1`] = `
<article
  className="near-places__place-card place-card"
  onMouseOver={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="near-places__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="/img/room.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          100
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            href="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": 40,
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        Canal View Prinsengracht
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;
